### PR TITLE
KAFKA-10060 GroupMetadataManager should not log if there are no offse…

### PR DESCRIPTION
**Pull request by chia7712**
address @ijuma [comment](https://issues.apache.org/jira/browse/KAFKA-10060?focusedCommentId=17118992&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17118992)
> I think the expired offsets one can be removed if there are no offsets to expire.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala [hunk 1]
```diff
@@ -786,7 +786,9 @@ class GroupMetadataManager(brokerId: Int,
       group.removeExpiredOffsets(currentTimestamp, config.offsetsRetentionMs)
     })
     offsetExpiredSensor.record(numOffsetsRemoved)
-    info(s"Removed $numOffsetsRemoved expired offsets in ${time.milliseconds() - currentTimestamp} milliseconds.")
+    if (numOffsetsRemoved > 0) {
+      info(s"Removed $numOffsetsRemoved expired offsets in ${time.milliseconds() - currentTimestamp} milliseconds.")
+    }
```
Comment (ijuma) [inline core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala]: Nit: no need for braces.

File: core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala [hunk 2]
```diff
@@ -85,21 +82,47 @@ class GroupMetadataManagerTest {
...
+    groupMetadataManager = new GroupMetadataManager(0, ApiVersion.latestVersion, offsetConfig, replicaManager,
+      mockKafkaZkClient, time, metrics)
     partition = EasyMock.niceMock(classOf[Partition])
   }
 
+  @Test
+  def testLogInfoFromCleanupGroupMetadata(): Unit = {
+    var expiredOffsets: Int = 0
+    var infoCount = 0
+    val gmm = new GroupMetadataManager(0, ApiVersion.latestVersion, offsetConfig, replicaManager, mockKafkaZkClient, time, metrics) {
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala:110]: create another GroupMetadataManager to override the methods we want to verify

File: core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala [hunk 3]
```diff
@@ -85,21 +82,47 @@ class GroupMetadataManagerTest {
       offsetsTopicCompressionCodec = config.offsetsTopicCompressionCodec,
       offsetCommitTimeoutMs = config.offsetCommitTimeoutMs,
       offsetCommitRequiredAcks = config.offsetCommitRequiredAcks)
+  }
 
-    defaultOffsetRetentionMs = offsetConfig.offsetsRetentionMs
-
+  private[this] def mockKafkaZkClient: KafkaZkClient = {
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala]: Nit: we don't use `private[this]` for tests. It's mostly a micro-optimization is not particularly useful for tests (`private` provides good enough semantics).

**Review by ijuma — COMMENTED** [review]
Changes look good generally, just a couple of nits.

File: core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala (hunk 1 above)
Comment (chia7712) [reply core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala]: done

File: core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala (hunk 3 above)
Comment (chia7712) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala]: done

**Comment by ijuma** [comment]
Unrelated flaky test failure:

`kafka.api.PlaintextAdminIntegrationTest.testAlterReplicaLogDirs`

**Review by ijuma — APPROVED** [review]
LGTM, thanks.

**Pull request merged**